### ISSUE TITLE
Is it possible to continue training a previous dataset?

### ISSUE THREAD

**Issue by VisionaryMind** [darglein/ADOP#75]
I have looked at the code, and the NeuralTrainer creates a datetime-stamped experiment folder upon each run. Would it be possible for me to add an additional config item to feed in the path to an existing folder, read the epoch folderset, and use the latest epoch in that folder to continue training? I have not yet tried this, as I want to make sure there are no other nuances here that need to be considered. Some datasets were stopped at earlier iterations, and it would be convenient to start where they left off, so as not to run everything from Epoch 0 again.

**Comment by VisionaryMind** [comment]
UPDATE: I went ahead and made the proposed changes above. Training appears to resume successfully from the last checkpoint.  I will go ahead and close out this issue.

**Comment by Gatsby23** [comment]
> UPDATE: I went ahead and made the proposed changes above. Training appears to resume successfully from the last checkpoint. I will go ahead and close out this issue.

Sorry for troubing you. How do you solve this problem ?

**Comment by VisionaryMind** [comment]
I am re-opening this issue, because after looking at the loss history, the pipeline appears to not be resuming, after all. It's loss values reset, regardless of providing an experiment directory / epoch #. @darglein , perhaps you could shed some light on other code components that must be modified to allow this feature. I am sure if I stared at the code long enough, a solution would materialize, but you are the author, and you know best.

**Comment by VisionaryMind** [comment]
I have attempted to use the LoadCheckpoint method in Pipeline.h prior to resuming training, while also setting the checkpoint_directory value in the config .ini, but it does not appear to work. Loss values jump back up to the top on resume:

96        ,378.884        ,0              ,0.0597312      ,0              ,0              ,0.000186146    ,136            
97        ,373.473        ,0              ,0.0586172      ,0              ,0              ,0.000174147    ,136            
98        ,374.662        ,0              ,0.0591767      ,0              ,0              ,0.000160695    ,136            
99        ,370.11         ,0              ,0.0565956      ,0              ,0              ,0.000159027    ,136            
100       ,375.5          ,0              ,0.0592704      ,0              ,0              ,0.000139422    ,136
101       ,850.828        ,0              ,0.19222        ,0              ,0              ,0.000399652    ,136            
102       ,683.594        ,0              ,0.145169       ,0              ,0              ,0.00065426     ,136            
103       ,604.459        ,0              ,0.122744       ,0              ,0              ,0.000813606    ,136    

I will continue to look for options here. As it stands, this pipeline will not work in any situation where there are limits on training time or the need to partition ML tasks (as we do in our lab).

**Comment by VisionaryMind** [comment]
After exhaustive experimentation, it appears the way to resume training from a previous checkpoint is much simpler than I had originally presumed. All you need to do is change the `checkpoint_directory `setting in the corresponding configuration file (i.e. path to the render_net.pth from the ADOP root). No need to bother with any other code changes. Keep in mind, epochs will reset back to 1, so if you have, for example, trained 400 epochs and want to run 200 more, change the `num_epochs` setting in the .ini to 200.
